src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/interfaces/ILobby.java

```diff
@@ -57,4 +57,12 @@ public interface ILobby extends Iterable<IPlayer>{
     IPlayer generatePlayer();
 
     void addPlayer(IPlayer player) throws FullLobbyException;
+
+    int getLobbyCapacity();
+
+    void reduceLobbyCapacity();
+
+    void setAllPlayersNotReady();
+
+    void balanceTeams();
```

Comment:
FYI:
I know, we have a lot of them and I don't like them but I cannot (and don't want to) refactor so much anymore.
https://dzone.com/articles/void-methods-considered-anti-pattern

Could you maybe let the reduceLobbyCapacity return an int (the new Capacity), setAllPlayersNotReady and balanceTeams boolean to indicate if the action was successful.